gear/gear/database.py

```diff
@@ -51,18 +52,61 @@ def transformer(fun):
     return await acontext_manager.__aexit__(exc_type, exc_val, exc_tb)
 
 
+def ssl_context_from_sql_config(sql_config):
+    ssl_mode = sql_config.get('ssl-mode', 'DISABLED')
+    if ssl_mode == 'DISABLED':
+        # change to the following in a follow up PR
+        # raise ValueError(f'cleartext database connections are not '
+        #                  f'permitted. {json.dumps(sql_config)}')
+        log.warn(f'using a cleartext mysql connection')
+
+        return False
+    elif ssl_mode == 'REQUIRED':
+        # change to the following in a follow up PR
+        # raise ValueError(f'unverified database connections are not '
+        #                  f'permitted. {json.dumps(sql_config)}')
+        assert 'ssl-cert' in sql_config
+        assert 'ssl-key' in sql_config
+
+        log.warn(f'not verifying msyql server certificates')
+
+        ssl_context = ssl.create_default_context()
+        ssl_context.load_cert_chain(
+            sql_config['ssl-cert'], keyfile=sql_config['ssl-key'], password=None)
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+        return ssl_context
+    elif ssl_mode == 'VERIFY_CA':
+        assert 'ssl-cert' in sql_config
+        assert 'ssl-key' in sql_config
+        assert 'ssl-ca' in sql_config
+
+        log.info(f'verifying msyql server certificates')
+
+        ssl_context = ssl.create_default_context(cafile=sql_config['ssl-ca'])
+        ssl_context.load_cert_chain(
+            sql_config['ssl-cert'], keyfile=sql_config['ssl-key'], password=None)
+        ssl_context.check_hostname = False
+        return ssl_context
+    else:
+        raise ValueError(f'only DISABLED, REQURIED, and VERIFY_CA are '
+                         'supported for ssl-mode. {json.dumps(sql_config)}')
+
+
 @retry_transient_mysql_errors
 async def create_database_pool(config_file=None, autocommit=True, maxsize=10):
     if config_file is None:
         config_file = os.environ.get('HAIL_DATABASE_CONFIG_FILE', '/sql-config/sql-config.json')
     with open(config_file, 'r') as f:
         sql_config = json.loads(f.read())
+    ssl_context = ssl_context_from_sql_config(sql_config)
+    assert ssl_context is not None
```

Comment:
Because TLS is so absolutely critical to safety, I want to ensure that there is never a bug in `ssl_context_from_sql_config` now or in the future that could pass a value `None` below. I don't know how these libraries interpret `None` and I don't want to take the chance that a bug somewhere else results in unencrypted connections.